R/run-log.R

```diff
@@ -46,13 +46,20 @@ run_log <- function(.base_dir, .recurse = TRUE) {
 #' @importFrom purrr map_lgl map compact
 #' @importFrom fs dir_ls
 #' @keywords internal
-find_models <- function(.base_dir, .recurse) {
+find_models <- function(.base_dir, .recurse , .filter = 0:0) {
 
   # get yaml files
   yaml_files <- dir_ls(.base_dir, recurse = .recurse)
+  yaml_files <- subset(yaml_files, !(yaml_files %>% basename() %>% stringr::str_remove(".yaml") %in% {{.filter}}))
   yaml_files <- str_subset(yaml_files, "\\.ya?ml$")
   yaml_files <- str_subset(yaml_files, "bbi\\.ya?ml$", negate = TRUE)
 
+
+  if(length(yaml_files) == 0)
+  {
+    warning("All models not selected by filter. Please adjust filter vector" %>% glue::glue())
```

Comment:
> For instance, when no YAML is found it will raise a warning and not abort in the find_models function.

Hmm, all right.  The other aspect of abort is that it stops execution, but I guess all of the downstream code is harmless with an empty set, so that should be fine to let go through (rather than, say, warn and return).